nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -0,0 +1,71 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.keycloak;
+in {
+  options = {
+    services.keycloak = {
+      enable = mkEnableOption "Keycloak identity and access server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keycloak;
+        defaultText = "pkgs.keycloak";
+        example = literalExample "pkgs.keycloak";
+        description = "Package that should be used for keycloak";
+      };
+
+      jbossBaseDir = mkOption {
```

Comment:
I'm under the impression configuration can be immutable and `keycloak` will operate fine that way. Is there a way we can separate configuration into an immutable directory (like `/nix/store/...`)? It might be nice to specify a separate logs directory as well (like `/var/log/keycloak`).

Reply:
When you say configuration can be immutable, what level of configuration are you thinking of?
Most usage supposes management of realms/clients/users/etc dynamically- are you thinking of the files listed above?

Reply:
Is there no concept of `root` owned configuration under `/etc` that isn't modified at runtime? While some `java` apps follow this pattern many do not... but I thought there was at least some concept of configuration (not state) with `keycloak` like this. I defer to your knowledge as I don't know much about the application.